tex/generic/pgf/math/pgfmathparser.code.tex

```diff
@@ -235,6 +235,16 @@
   \pgfmath@tokens@make{dimexpr}{\dimexpr\glueexpr}
 \fi
 \pgfmath@tokens@make{unit}{{bp}{cc}{cm}{dd}{em}{ex}{in}{mm}{pc}{pt}{sp}}
+\ifx\pdftexversion\@undefined\else % for pdfTeX
```

Comment:
Since we require e-TeX you can also use `\ifdefined`.
```suggestion
\ifdefined\pdftexversion % for pdfTeX
```

Reply:
If e-TeX is actually required, you are right, but a few lines above that line I've added says

```
\ifx\dimexpr\@undefined\else % needed for classic tex
```

so I guess this file should be intended to work also with both classix tex.

Reply:
Ah, right.  I think `pgfmath` technically does not require e-TeX yet.  However, this made me thinking whether we should just use `iftex` for engine detection.

Reply:
`iftex` started to support detecting ptex only recently (2019 by David), so it may be safer to keep my implementation as-is.